python/cudf/cudf/core/column/column.py

```diff
@@ -80,21 +80,23 @@ def data_array_view(self):
         else:
             dtype = self.dtype
 
-        result = rmm.device_array_from_ptr(
-            ptr=self.data.ptr, nelem=len(self), dtype=dtype
+        result = cuda.as_cuda_array(self.data)
+        # Workaround until `.view(...)` can change itemsize
+        # xref: https://github.com/numba/numba/issues/4829
+        result = cuda.devicearray.DeviceNDArray(
+            shape=(result.nbytes // dtype.itemsize,),
+            strides=(dtype.itemsize,),
+            dtype=dtype,
+            gpu_data=result.gpu_data,
         )
-        result.gpu_data._obj = self
         return result
 
     @property
     def mask_array_view(self):
         """
         View the mask as a device array
         """
-        result = rmm.device_array_from_ptr(
-            ptr=self.mask.ptr, nelem=self.mask.size, dtype=np.int8,
-        )
-        result.gpu_data._obj = self
+        result = cuda.as_cuda_array(self.mask).view(np.int8)
```

Comment:
Same here

Reply:
Same answer [as above]( https://github.com/rapidsai/cudf/pull/4101#discussion_r376728920 ) excluding the workaround.